torch/_dynamo/variables/builder.py

```diff
@@ -1716,16 +1716,24 @@ def wrap_to_fake_tensor_and_record(
         or is_traceable_wrapper_subclass(e)
     ):
         assert source is not None
+
         static_shapes, reason = tensor_always_has_static_shape(
             e, is_tensor, guard_source=source.guard_source()
         )
 
+        widr = WeakIdRef(e)
         dynamic_dims, constraint_dims = None, None
-        if not e.is_nested:
-            # TODO: We should probably support this for nested tensors too
-            dynamic_dims, constraint_dims = _automatic_dynamic(
-                e, tx, source.name(), static_shapes
-            )
+        # See Note - [On fake tensor policy and fresh fake modes for backends]
+        if widr in TracingContext.get().weak_tensor_ref_to_fakification_policy:
```

Comment:
Curiosity q: when does this happen? My understanding (probably wrong) was that this codepath runs whenever we create a fresh fake tensor during dynamo tracing, so I'm surprised that we might already have a cached policy for it here.

Reply:
Let me find the exact codepath, iirc this happens due to aliasing + our lazy VT approach